app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryCreateActivity.kt

```diff
@@ -11,68 +14,122 @@ import android.annotation.SuppressLint
 
+    // Location
     lateinit var locationComponentManager: LocationComponentManager
 
+    // Area
+    private var longClickConsumed = false
+    private lateinit var areaBuilder: AreaBuilder
+    private lateinit var areaBuilderDrawer: MapboxAreaBuilderDrawer
+
+    // Drawer
+    private lateinit var missionDrawer: MapboxMissionDrawer
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         Mapbox.getInstance(this, getString(R.string.mapbox_access_token))
-
         setContentView(R.layout.activity_itinerary_create)
 
-        //Create a "back button" in the action bar up
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
-
         mapView = findViewById(R.id.mapView)
         mapView.onCreate(savedInstanceState)
-        mapView.getMapAsync { mapboxMap ->
 
-            locationComponentManager = LocationComponentManager(this, mapboxMap)
-            mapboxMap.setStyle(Style.MAPBOX_STREETS) {
-                // Map is set up and the style has loaded. Now we can add data or make other map adjustments
+        mapView.getMapAsync(this)
+        mapView.contentDescription = getString(R.string.map_not_ready)
 
-                locationComponentManager.enableLocationComponent(it)
 
-            }
-        }
+        //Create a "back button" in the action bar up
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
         goToSaveButton = findViewById(R.id.buttonToSaveActivity)
         goToSaveButton.isEnabled = authService.hasActiveSession()
     }
 
     fun goToSaveActivity(@Suppress("UNUSED_PARAMETER") view: View) {
+
+        // TODO Replace by the actual MappingMission flight path once we are able to generate it from an area
+        if (areaBuilder.getShapeVertices() != null){
+            flightPath = ArrayList(areaBuilder.getShapeVertices());
+        }
+
         val intent = Intent(this, SaveMappingMissionActivity::class.java)
         intent.putExtra("flightPath", flightPath)
         startActivity(intent)
     }
 
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        locationComponentManager = LocationComponentManager(this, mapboxMap)
+
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            locationComponentManager.enableLocationComponent(style)
+
+            fun onLongClickConsumed(): Boolean {
+                longClickConsumed = true;
+                return false
+            }
+
+            areaBuilderDrawer =
+                MapboxAreaBuilderDrawer(mapView, mapboxMap, style) { onLongClickConsumed() }
+            missionDrawer = MapboxMissionDrawer(mapView, mapboxMap, style)
+
+            mapboxMap.addOnMapClickListener(this)
+            mapboxMap.addOnMapLongClickListener(this)
+
+            areaBuilder = PolygonBuilder()
+            //areaBuilder = ParallelogramBuilder()
+
+            //areaBuilder.onAreaChanged.add { missionBuilder.withSearchArea(it) }
```

Comment:
Are these comments still needed ?

Reply:
Yes, same as above for the first one. For the second one, we will able to uncomment it as soon as we are able te generate a mapping mission from the area, for now I used the border of the area as flightpath.